### DISCUSSION TITLE
SitemapGenerator::create() does not work on shared host

### DISCUSSION THREAD

**Discussion by babak271** [spatie/laravel-sitemap#359] [Q&A]
I've created sitemap using `SitemapGenerator::create(config('app.url'))` in local and it works flawlessly, but when I want to create it in global site which is deployed on a shared host, it cannot create any `tags` and the result looks like below:
```
<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml">
</urlset>
```
I've checked every piece of configurations for thousands of times, but I couldn't figure out the problem.
I've also checked [issue210](https://github.com/spatie/laravel-sitemap/issues/210) and I've deleted `robots.txt`, but it didn't solve the problem.

**Comment by BryantChi** [comment]
if your `config('app.url')` is `https`, you should be add `RequestOptions::VERIFY => false` at `config/sitemap.php`.
I hope this can help you.

**Comment by schonhoff** [comment]
That helped me a lot thank you @BryantChi

**Comment by babak271** [comment]
> RequestOptions::VERIFY => false

I have added `RequestOptions::VERIFY => false`, but it did not work on a direcadmin shared host. I have tested it both on *http* and *https* sites, didn't work.

**Comment by connor11528** [comment]
I am also having the same issue. The package generates an empty sitemap for my app hosted on [Render](https://render.com/docs/deploy-php-laravel-docker)

**Comment by gaetan-hexadog** [comment]
Same here: empty sitemap on shared hosting (ovh). I tried the `RequestOptions::VERIFY => false` option without any success.
The package is working on local environment but once sent to production it generates an empty sitemap.

**Comment by connor11528** [comment]
what helped for me was adding `www` in the URL string. My site autocorrects to include the www: https://employbl.com

**Comment by gaetan-hexadog** [comment]
Thanks for the hint  but there is no difference on my side with or without `www` in the URL string.

**Comment by babak271** [comment]
@freekmurze Are you going to open this issue? It remains unsolved for us.

**Comment by fdjkgh580** [comment]
Hey guys, you can try it. see the answer for this comment.
https://github.com/spatie/laravel-sitemap/discussions/505#discussioncomment-7299333
